internal/command/deploy/machines.go

```diff
@@ -539,19 +617,33 @@ func (md *machineDeployment) resolveUpdatedMachineConfig(origMachineRaw *api.Mac
 			Config: &api.MachineConfig{},
 		}
 	}
+	newConfig := machine.CloneConfig(origMachineRaw.Config)
+	if newConfig.Metadata == nil {
+		newConfig.Metadata = map[string]string{}
+	}
```

Comment:
wdyt if this is moved to line 634 (right before setting metadata), it can simplify that code too and remove a nested if.